tinygrad/codegen/uopgraph.py

```diff
@@ -312,6 +312,8 @@ def index_collapse(idx,rng,buf,add,mul,ld,reduce):
   ((NOp.var("x") / NOp.var("x2")) / NOp.var("x3"), lambda x,x2,x3: x/(x2*x3)), # (x/x2)/x3 -> x/(x2*x3)
   (-(NOp.var("x") + NOp.var("y")), lambda x,y: (-x)+(-y)),  # -(x+y) -> -x + -y
   ((NOp.cvar("c0") + NOp.var("x")).lt(NOp.cvar("c1")), lambda x,c0,c1: UOp.lt(x, c1-c0)),  # c0 + x < c1 -> x < c1 - c0
+  ((NOp.var("x") & NOp.var("x")), lambda x: x),
+  ((NOp.var("x") | NOp.var("x")), lambda x: x),
```

Comment:
can be its own diff.

Reply:
removing these lines, redundant re: #6412